### PR TITLE
Fix bug with handling orders containing deleted variants in 3.1

### PR DESCRIPTION
**Pull request by jakubkuc**
I want to merge this change because user should be able to manage orders (adding tracking number, deleting) containing variants that were deleted after the order was created.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4253 Errors when handling orders containing deleted variants](https://app.clickup.com/t/2549495/SALEOR-4253)

**Review by IKarbowiak — APPROVED** [review]

**Comment by korycins** [comment]
I believe this should be merged to master instead of 3.1 branch

File: saleor/order/tests/test_notifications.py [hunk 1]
```diff
@@ -160,6 +160,46 @@ def test_get_order_line_payload(order_line):
...
+    order_line.variant = None
+    payload = get_order_line_payload(order_line)
+    unit_tax_amount = (
+        order_line.unit_price_gross_amount - order_line.unit_price_net_amount
+    )
+    total_gross = order_line.unit_price_gross * order_line.quantity
+    total_net = order_line.unit_price_net * order_line.quantity
+    total_tax = total_gross - total_net
+    assert payload == {
+        "variant": None,
```
Comment (korycins) [inline saleor/order/tests/test_notifications.py:173]: IMHO the standard payload is covered in different test. I would say that this is enough to only check if a variant field exists in the payload. (No need to twice check the same payload)

Comment (jakubkuc) [reply saleor/order/tests/test_notifications.py:173]: It was also my concern. But I think other tests do not cover the payload with deleted variant so we can't be sure if any change made in code won't change anything in the whole payload in this specific case. What do you think?

**Review by korycins — APPROVED** [review]

**Comment by jakubkuc** [comment]
Invalid pull request base

**Pull request closed without merging**